[AIRFLOW-5663] Switch to real-time logging in PythonVirtualenvOperator

wjiangqc opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5663

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Allow us to see the logs while the operator is running.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Existing 19 unit tests passed.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

Commit:
[AIRFLOW-5663]: Switch to real-time logging in PythonVirtualenvOperator.

mik-laj commented on airflow/operators/python_operator.py:
```diff
@@ -330,16 +330,20 @@ def _pass_op_args(self):
+        self.log.info("Got output\n")
+        with self._sp.stdout:
+            for line in iter(self._sp.stdout.readline, b''):
+                self.log.info(line)
```
```suggestion
                self.log.info("%s", line)
```
Data should be passed as arguments so that they are written in a different color.

wjiangqc committed:
Update airflow/operators/python_operator.py

Resolve comment: "Data should be passed as arguments so that they are written in a different color."

Co-Authored-By: Kamil Breguła <mik-laj@users.noreply.github.com>

wjiangqc committed:
Raise exception for negative returncode.

mik-laj commented on airflow/operators/python_operator.py:
```diff
@@ -330,16 +330,20 @@ def _pass_op_args(self):
+                                    stderr=subprocess.STDOUT,
+                                    bufsize=0,
+                                    close_fds=True)
+        self.log.info("Got output\n")
```
```suggestion
        self.log.info("Got output")
```
Each log message is printed in a separated line.

mik-laj commented on airflow/operators/python_operator.py:
```diff
@@ -330,16 +330,20 @@ def _pass_op_args(self):
+        self.log.info("Got output\n")
+        with self._sp.stdout:
+            for line in iter(self._sp.stdout.readline, b''):
+                self.log.info("%s", line)
```
```suggestion
                self.log.info("%s", line.decode().rstrip())
```
This will make the logs clearer.

wjiangqc committed:
Update airflow/operators/python_operator.py

Resolved comment: Each log message is printed in a separated line.

Co-Authored-By: Kamil Breguła <mik-laj@users.noreply.github.com>

wjiangqc committed:
Update airflow/operators/python_operator.py

Resolved comment: This will make the logs clearer.

Co-Authored-By: Kamil Breguła <mik-laj@users.noreply.github.com>
